lib/restforce/concerns/api.rb

```diff
@@ -364,6 +364,9 @@ def upsert!(sobject, field, attrs)
         attrs = attrs.dup
         external_id =
           extract_case_insensitive_string_or_symbol_key_from_hash!(attrs, field)
+        if field.to_s != "Id" && (external_id.nil? || external_id.strip.empty?)
+          raise ArgumentError, 'External id field missing from attrs.'
```

Comment:
Can we change this to "Specified external ID field missing from provided attributes"?